pkg/controller/periodic/periodic.go

```diff
@@ -98,10 +108,23 @@ func (c *Controller) Gather() {
 		Steps:    threshold,
 		Cap:      interval,
 	}
-	for name := range c.gatherers {
+
+	// flush when all necessary gatherers were processed
+	defer func() {
+		if err := c.recorder.Flush(); err != nil {
+			klog.Errorf("Unable to flush the recorder: %v", err)
+		}
+	}()
+
+	for periodMultiplier, gatherer := range c.gatherers {
```

Comment:
I don't think I am in favor of this `periodMultiplier`. I think it's not so clear and maybe can bring some unexpected consequences. We just want to run the workloads gatherer say once a day and I think the period should be ideally independent. I think running something like:

`go wait.Until(func() { c.periodicTrigger(stopCh) }, time.Second, stopCh)` 

with different period (defined inside the `periodicTrigger`) would be more appropriate.

Reply:
But then we are back to having different archives with and without clusterconfig gatherer

Reply:
I don't think so. I think it's just about when you call recorder `Flush` and you already share it (few lines above) for all the gatherers right.

Reply:
If we have multiple goroutines doing gathering we'd also need to make sure there's no syncronization problems when they run at the same time.

Reply:
Yes probably yes